src/query_plan/query_planner.rs

```diff
@@ -140,11 +146,12 @@ pub struct QueryPlanner {
     abstract_types_with_inconsistent_runtime_types: IndexSet<AbstractTypeDefinitionPosition>,
     // TODO: Port _lastGeneratedPlanStatistics from the JS codebase in a way that keeps QueryPlanner
     // immutable.
+    // last_generated_plan_statistics: QueryPlanningStatistics,
 }
 
 impl QueryPlanner {
     pub fn new(
-        supergraph: Supergraph,
+        supergraph: &Supergraph,
```

Comment:
This doesn't need ownership as it just copies the things it needs off the supergraph into the query planner struct.